pkg/store/mockstore/unistore/cophandler/closure_exec.go

```diff
@@ -932,8 +925,16 @@ func (e *closureExecutor) indexScanProcessCore(key, value []byte) error {
 	}
 	// Add ExtraPhysTblID if requested
 	// Assumes it is always last!
-	if e.columnInfos[len(e.columnInfos)-1].ColumnId == model.ExtraPhysTblID {
-		tblID := tablecodec.DecodeTableID(key)
+	if e.columnInfos[len(e.columnInfos)-1].ColumnId == model.ExtraPhysTblID && min(len(e.fieldTps), len(values)) < len(e.columnInfos) {
```

Comment:
Is the `min(...) < len(...)` added in the case where the Partition ID is already returned as a column, like where it previously would have had ExtraPidColID?

Reply:
Updated, maybe you can get the answer with the new comments in the code